addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -5677,6 +5677,13 @@ const ImageHandlerOption = SnippetOptionWidget.extend({
      */
     async _loadImageInfo(attachmentSrc = '') {
         const img = this._getImg();
+        // Remove the `original` attributes if there is a shape, as they are
+        // blocking if they are not correctly hardcoded.
+        if (img.dataset.shape) {
+            delete img.dataset.originalId;
+            delete img.dataset.originalSrc;
+            delete img.dataset.originalMimetype;
+        }
```

Comment:
@sobo-odoo Questions about this:

- Do you have an example where we "not correctly hardcoded" such info? There does not seem to be a linked PR where such wrong hardcoding is being fixed in themes?
- I don't understand how the wrong hardcoding is related to the fact a shape is used
- Your commit message does not seem to imply that this occurs since a recent change, so should not it target something below 16.0?
- Are those info never useful when there is a shape?
    - If yes, why are not they removed when a shape is added?
    - If no (as I think it is), should not we try to use them and instead detect the wrong hardcoding somehow? Typically, if I save a custom snippet via the save button in the right panel and that snippet contains a shape with a an image, for example a snippet with the 4 board members of my company using a shape on their picture... would not the saved dataset be correct?

EDIT: I saw the second commit afterwards. So it's only the mass mailing templates where there are such problems? Only in 16.0? If yes:
- Squash both commits: see I did not understand what this was about without seeing the second commit, both things should be fixed at the same time and allows to explain things more clearly :+1: 
- That question remains: "I don't understand how the wrong hardcoding is related to the fact a shape is used" -> = does not the code fails too if there is no shape and a wrong hardcoded link? In the end, I wonder if this should not be about patching the shape code (not the loading image info code) to say "if the original src is one of these wrong hardcoded links that we made, just remove it", then remove that code in master ?

> It finally blocks when it arrives in the `_computeAvailableWidths` method, where it awaits endlessly.

=> Why does it await endlessly? Should not that be fixed to make the code more robust and maybe even allow to log errors so we can detect wrong hardcoded xml directly? :)

Reply:
@qsm-odoo  
> Why does it await endlessly? 

In fact, it doesn't await endlessly, it's just that there is no catch in case of a `reject` (when the file doesn't exist) and so the code after the `await` is never executed. Therefore, the start of the options never terminates. (Just writing here what I told last week :wink: ) 

> does not the code fails too if there is no shape and a wrong hardcoded link?

Indeed, it can fail even without shapes. In fact, I think it depends on the presence of a `data-mimetype` attribute (which is checked before applying the ImageTools option (shape, filter,...)) => If the links are wrong and if there is a supported mimetype, then it fails. 

From what I saw, there are two cases to consider: 
1. src is in the form of `/web/image/...` => if the "link" is wrong, the missing image is automatically replaced by a placeholder (see the `/web/image` route in `web/controllers/binary.py`). However, I just saw (as I am writing this...) that the bug still happens if there is a mimetype, as the link is still the wrong one :grimacing: I will investigate.

2. src is in another form: if the link is wrong, nothing replaces the image (and so nothing is displayed). Firstly, it is not easy to fix the image because since nothing is displayed, we can't click to make the image options appear and replace it (we need to modify the DOM manually). Secondly, this is the case where the await rejects are not caught properly. 

For the case 2, I added a check of the `src` and `originalSrc` before the start of the option (see the code I force pushed :slightly_smiling_face: ). I thought it was a good place to check as it will cover all the cases (I think it is better than try catching every problematic await (there are at least 2, depending on the situation)). 

 I will investigate case 1, as I just saw it :grimacing:

Reply:
Case 1 happens because even though there is a content (the placeholder), there is apparently no attachment linked to the incorrect src and so `loadImageInfo` (in image_processing.js) does not set the `data-original-*` attributes (because it cannot find the image (original = false)). And so, since there is no `data-original-src`, the await in `applyModifications` (in the same file) is rejected (but there is no catch to manage it ). 
So I added a check if the attribute is there before calling `applyModifications`. I removed the mimetype so the shape, filter,... options are not displayed (since they won't work anyway).